storybookjs/react-native
# chore: tweak navigator design to match SB web navigator design

## Issue:

The old design was a bit unorganized and doesn't look like the Storybook web interface, so I tried to make it as close to it as possible.

## What I did

I tweaked the navigator design to make it look like the one in the Storybook web interface.

Here's how it looks after the changes:

![Storybook react-native navigator redesigned](https://user-images.githubusercontent.com/68241710/153572256-36e3a101-d662-4e68-9cd8-0668c7f3da28.png)

## How to test

You just need to run the example app and see the changes in action.

- Does this need a new example in examples/native? No
- Does this need an update to the documentation? No

> NOTE: There're some issues with `@emotion/native` and TypeScript because I'm using the `transform` property which uses an array of objects in react-native instead of a string. If you have any recommendations on how to fix it please let me know.

> NOTE: I didn't use images/SVGs for the grid icon, I did it with react-native views just to not include images or add dependency of react-native-svg to the package.

## What's next?

I'll work on improving it in my free time, things I want to add:

- Toggle component stories.
- Add a bookmark icon beside the story name (like in the web version).
- Add a search icon in the search input.
- Improve the a11y of the navigator as a whole.
- Focus on search input on <kbd>/</kbd> press.
